tests/test_config.py

```diff
@@ -94,3 +97,13 @@ def test_from_sys_argv(config_dict: dict, cli_args: list, monkeypatch: pytest.Mo
 def assert_config_correct(config_dict: dict, config_obj: Config) -> None:
     for key, item in config_dict.items():
         assert getattr(config_obj, key.replace("-", "_")) == item
+
+
+def test_generate_yaml_from_parser() -> None:
+    config_obj = Config.from_args("--generate-yaml", "--file", "testfile.yaml")
+    tmp_file = NamedTemporaryFile().name
```

Comment:
Just noticed this,  we should probably use the built in pytest fixture rather than including a separate package.  But if that isn't trivial, leave it for now.